src/MatrixFields/var_set.jl

```diff
@@ -0,0 +1,336 @@
+const VariableNameTup = NTuple{2, VariableName}
+
+"""
+    VarSet{T}(values, [var_tree])
+
+Implemented subtypes:
+    - `FieldVectorKeys`
+    - `FieldMatrixKeys`
+
+TODO
+"""
+struct VarSet{
+    T <: Union{VariableName, VariableNameTup},
+    V <: NTuple{<:Any, T},
+    VT <: Union{VarTree, Nothing},
+} <: AbstractSet{T}
```

Comment:
how (and where) is a `VarSet` used?

Reply:
It's the equivalent of the built-in `KeySet` for `VarDict`, but with additional functionality for handling `VariableName`s and row-column pairs of `VariableNames`.

Reply:
I added a blurb about each of the new structs in the PR description.